src/smt/theory_str_noodler/theory_str_noodler.cpp

```diff
@@ -392,15 +392,21 @@ namespace smt::noodler {
         } else if(m_util_s.str.is_replace_re_all(n)) { // str.replace_re_all
             util::throw_error("str.replace_re_all is not supported");
         } else if (m_util_s.str.is_is_digit(n)) { // str.is_digit
-            util::throw_error("str.is_digit is not supported");
-        } else if (m_util_s.str.is_to_code(n)) { // str.to_code
-            util::throw_error("str.to_code is not supported");
-        } else if (m_util_s.str.is_from_code(n)) { // str.from_code
-            util::throw_error("str.from_code is not supported");
-        } else if (m_util_s.str.is_stoi(n)) { // str.to_int
-            util::throw_error("str.to_int is not supported");
-        } else if (m_util_s.str.is_itos(n)) { // str.from_int
-            util::throw_error("str.from_int is not supported");
+            handle_is_digit(n);
+        } else if (
+            m_util_s.str.is_stoi(n) || // str.to_int
+            m_util_s.str.is_itos(n) // str.from_int
+        ) {
+            // handle_transform can handle to/from_int, but decision procedure cannot.
+            // We throw error here so that we get to unknown faster. After decision
+            // procedure gets support for it, remove this and let it fall trough with
+            // is/from_code to handle_transform
+            util::throw_error("str.to_int and str.from_int is not supported (yet)");
+        } else if (
+            m_util_s.str.is_to_code(n) || // str.to_code
+            m_util_s.str.is_from_code(n) // str.from_code
+        ) {
+            handle_transform(n);
```

Comment:
You something containing code instead `handle_transform`? It sounds like too general.

Reply:
It also handles `to_int` and `from_int`. But I also do not like the name `transform`, but I cannot find a better name.

Reply:
what about `handle_int_conversion`?

Reply:
Maybe just conversions? I would not say `to_code` and `from_code` are int conversions

Reply:
ok

Reply:
I renamed it (+ everything related) to use convert instead of transfom.